### PR TITLE
Switch to nlohmann/json from json11

### PR DESCRIPTION
**Pull request by mtribiere**
**Description**

This is a work in progress to switch to a new JSON library. From json11 to nlohmann/json.
#24523 

**Verification**
I do not posses any hardware so my testing capabilities are quite limited. However the CI and Unit test can give a good indicator. 
Some tests in the driving simulator might also be a good idea.

### CODE REVIEW

File: common/tests/test_swaglog.cc [hunk 1]
```diff
@@ -39,26 +39,25 @@ void recv_log(int thread_cnt, int thread_msg_cnt) {
 
     REQUIRE(buf[0] == CLOUDLOG_DEBUG);
```
Comment (mtribiere) [inline common/tests/test_swaglog.cc:40]: This test fail, but I'm not sure why it's here in the first place. If you got a JSON as a string, then buf[0] should be '{' right ?

Comment (adeebshihadeh) [reply common/tests/test_swaglog.cc:40]: see https://github.com/commaai/openpilot/blob/78e20cff3906ee76c356e8bdef920397e60389b4/common/swaglog.cc#L94

File: common/swaglog.cc [hunk 2]
```diff
@@ -92,7 +92,7 @@ static void cloudlog_common(int levelnum, const char* filename, int lineno, cons
 
   std::string log_s;
   log_s += (char)levelnum;
-  ((json11::Json)log_j).dump(log_s);
+  log_s = log_j.dump();
```
Comment (adeebshihadeh) [inline common/swaglog.cc:95]: this should fix the test
```suggestion
  log_s += log_j.dump();
```

**Comment by adeebshihadeh** [comment]
> **Verification** I do not posses any hardware so my testing capabilities are quite limited. However the CI and Unit test can give a good indicator. Some tests in the driving simulator might also be a good idea.

This change should be well-tested by our CI. The last remaining json11 usage is in selfdrive/modeld/thneed. That doesn't build by default on PC, but you can build it with `scons --pc-thneed`. Once you get that building properly, I can test it.

**Comment by mtribiere** [comment]
Thanks, will do asap

**Comment by adeebshihadeh** [comment]
Thanks! Want to try #26739? It might be even easier than this one.

**Pull request merged**